src/MTransformationMatrix.inl

```diff
@@ -0,0 +1,322 @@
+            throw std::logic_error(status.errorString().asChar());
+        }
+
+        return result;
+    }, py::arg("space"), _doc_TransformationMatrix_rotatePivot)
+
+    .def("rotatePivotTranslation", [](MTransformationMatrix & self, MSpace::Space space) -> MVector {
+        MStatus status;
+        MVector result = self.rotatePivotTranslation(space, &status);
+
+        if (!status) {
+            throw std::logic_error(status.errorString().asChar());
+        }
+
+        return result;
+    }, py::arg("space"), _doc_TransformationMatrix_rotatePivotTranslation)
+
+    .def_property_readonly("quaternionRotation", &MTransformationMatrix::rotation,
+        "Returns the rotation component of the transformation matrix as a quaternion.")
+
+    .def_property_readonly("eulerRotation", &MTransformationMatrix::eulerRotation,
+        "Returns the rotation component of the transformation matrix as an euler rotation.")
+    
+    .def("setRotation", [](MTransformationMatrix & self, const MQuaternion& rotation) {
+        self.rotateTo(rotation);
+    }, py::arg("rotation"), "Set the rotation component of the transformation matrix using a quaternion.")
+
+    .def("setRotation", [](MTransformationMatrix & self, const MEulerRotation& rotation) {
+        self.rotateTo(rotation);
+    }, py::arg("rotation"), "Set the rotation component of the transformation matrix using an euler rotation.")
+
+    .def("rotateBy", [](MTransformationMatrix & self, const MQuaternion& rotate, MSpace::Space space) {
+        MStatus status;
+        self.rotateBy(rotate, space, &status);
+
+        if (!status) {
+            throw std::logic_error(status.errorString().asChar());
+        }
+    }, py::arg("rotate"), py::arg("space"), "Adds to the rotation component of the rotation matrix by rotating relative to the existing transformation using a quaternion.")
+
+    .def("rotateBy", [](MTransformationMatrix & self, const MEulerRotation& rotate, MSpace::Space space) {
+        MStatus status;
+        self.rotateBy(rotate, space, &status);
+
+        if (!status) {
+            throw std::logic_error(status.errorString().asChar());
+        }
+    }, py::arg("rotate"), py::arg("space"), "Adds to the rotation component of the rotation matrix by rotating relative to the existing transformation using an euler rotation.")
+
+    .def_property_readonly("rotationOrder", [](MTransformationMatrix & self) -> MTransformationMatrix::RotationOrder {
+        MStatus status;
+        auto result = self.rotationOrder(&status);
+
+        if (!status) {
+            throw std::logic_error(status.errorString().asChar());
+        }
+        
+        return result;
+    }, _doc_TransformationMatrix_rotationOrder)
+
+    .def("rotationOrientation", &MTransformationMatrix::rotationOrientation, _doc_TransformationMatrix_rotationOrientation)
+
+    .def("setRotationOrientation", &MTransformationMatrix::setRotationOrientation, _doc_TransformationMatrix_setRotationOrientation)
```

Comment:
You should add `py::arg` here as well.
I've named the argument `q` here to match Maya's documentation though I'm not super fond of these short names.

```suggestion
    .def("setRotationOrientation", &MTransformationMatrix::setRotationOrientation, py::arg("q"), _doc_TransformationMatrix_setRotationOrientation)
```